src/VirtualShield.cpp

```diff
@@ -420,12 +419,18 @@ void VirtualShield::onJsonStringReceived(char* json, ShieldEvent* shieldEvent) {
 /// <param name="length">The length.</param>
 /// <param name="shieldEvent">The shield event.</param>
 void VirtualShield::onStringReceived(char* buffer, int length, ShieldEvent* shieldEvent) {
+#if defined(_WINDOWS_)
+	char *json = new char[length + 1];
```

Comment:
For temporary local buffers, alloca might be a better option. It avoids thrashing the heap. Works on Arduino as well, so you don't need two happy paths